src/squlearn/kernel/ml/qgpc.py

```diff
@@ -53,10 +53,89 @@ class QGPC(GaussianProcessClassifier):
     --------
     """
 
-    def __init__(self, quantum_kernel: KernelMatrixBase, **kwargs):
+    def __init__(self, *, quantum_kernel: KernelMatrixBase, **kwargs) -> None:
         self._quantum_kernel = quantum_kernel
-        super().__init__(**kwargs)
-        self.kernel = kernel_wrapper(quantum_kernel)
+        # Apply kwargs to set_params of quantum kernel
+        valid_params_quantum_kernel = self._quantum_kernel.get_params(deep=True)
+        set_quantum_kernel_params_dict = {}
+        for key, value in kwargs.items():
+            if key in valid_params_quantum_kernel:
+                set_quantum_kernel_params_dict[key] = value
+
+        if len(set_quantum_kernel_params_dict) > 0:
+            self._quantum_kernel.set_params(**set_quantum_kernel_params_dict)
+
+        # remove quantum_kernel_kwargs for QGPC initialization
+        for key in set_quantum_kernel_params_dict:
+            kwargs.pop(key, None)
```

Comment:
This could be cleaner with e.g. bitwise and (`&`) on the keys, but as long as it works I don't mind changing it at a later point...